lib/dirac_mobius.cpp

```diff
@@ -372,4 +363,438 @@ namespace quda {
     deleteTmp(&tmp1, reset1);
   }
 
+  void DiracMobiusPC::MdagMLocal(ColorSpinorField &out, const ColorSpinorField &in) const
+  {
+    if (zMobius) { errorQuda("DiracMobiusPC::MdagMLocal doesn't currently support zMobius.\n"); }
+
+    checkDWF(in, out);
+    // checkParitySpinor(in, out);
+    checkSpinorAlias(in, out);
+
+    ColorSpinorParam csParam(out);
+    csParam.create = QUDA_NULL_FIELD_CREATE;
+
+    ColorSpinorField *unextended_tmp1 = ColorSpinorField::Create(csParam);
```

Comment:
We should create these fields once and persist it thereafter. Technically the pool allocator should keep this from being a problem, but it's still a better practice.

Reply:
So something like
```
static ColorSpinorField *unextended_tmp1 = ColorSpinorField::Create(csParam);
```
?
I am little bit uncomfortable about having these pointer staying there util the end of the app.

Reply:
This was benchmarked, and found not to be a problem.  

In general, beyond the pool allocation, allocating fields has an overhead when a field has communication.  By definition these fields have no communication, and the overhead of this reallocation has been tested to be zero.  There are persistent memory pointers in the dirac objects already, e.g., `tmp1` and `tmp2`, but there use needs to be cleaned up and that design is far from perfect.  Hence why this present design is sufficient for now, until we have an improved pool allocator for temporary fields like this.

Reply:
If it's benchmarked sounds good. I'd like a comment added on that, and a `FIXME`